samza-core/src/main/scala/org/apache/samza/checkpoint/CheckpointTool.scala

```diff
@@ -117,10 +118,15 @@ object CheckpointTool {
         .mapValues(m => m.reduce( _ ++ _))  // Merge all the maps of SSPs->Offset into one for the whole taskname
     }
 
-    override def loadConfig(options: OptionSet): MapConfig = {
+    override def loadConfig(options: OptionSet): Config = {
       val config = super.loadConfig(options)
       if (options.has(newOffsetsOpt)) {
-        val properties = configFactory.getConfig(options.valueOf(newOffsetsOpt))
+        val in = new FileInputStream(options.valueOf(newOffsetsOpt).getPath)
```

Comment:
nit: rename `in` to something meaningful?

Reply:
Renamed to newOffsetsInputStream